From af316f0cb4c4291152b5fd5a5560e7e73a0f1271 Mon Sep 17 00:00:00 2001
From: Scott Wood <scottwood@freescale.com>
Date: Thu, 16 May 2013 17:22:42 -0500
Subject: [PATCH 015/547] powerpc/e6500: Rename PPC_E6500_REV1_BUGS to
 FSL_ERRATUM_A_05337

Also make it default y so that people who don't update their config
won't get caught by the late config change.

Signed-off-by: Scott Wood <scottwood@freescale.com>
Change-Id: I28d5752dec9ce1c65e8d50dcbd9c2771d57184e2
Reviewed-on: http://git.am.freescale.net:8181/2559
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/powerpc/mm/tlb_nohash.c           |  2 +-
 arch/powerpc/platforms/Kconfig.cputype | 12 +++++++-----
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/powerpc/mm/tlb_nohash.c b/arch/powerpc/mm/tlb_nohash.c
index a6979a5..8e52b81 100644
--- a/arch/powerpc/mm/tlb_nohash.c
+++ b/arch/powerpc/mm/tlb_nohash.c
@@ -465,7 +465,7 @@ static void setup_page_sizes(void)
 		tlb1ps = mfspr(SPRN_TLB1PS);
 		eptcfg = mfspr(SPRN_EPTCFG);
 
-#ifndef CONFIG_PPC_E6500_REV1_BUGS
+#ifndef CONFIG_FSL_ERRATUM_A_005337
 		if ((tlb1cfg & TLBnCFG_IND) && (tlb0cfg & TLBnCFG_PT))
 			book3e_htw_mode = PPC_HTW_FSL;
 
diff --git a/arch/powerpc/platforms/Kconfig.cputype b/arch/powerpc/platforms/Kconfig.cputype
index 3c7016f..27be19c 100644
--- a/arch/powerpc/platforms/Kconfig.cputype
+++ b/arch/powerpc/platforms/Kconfig.cputype
@@ -164,14 +164,16 @@ config PPC_E500MC
 	  such as e5500/e6500), and must be disabled for running on
 	  e500v1 or e500v2.
 
-config PPC_E6500_REV1_BUGS
-	bool "Work around e6500 rev1 bugs"
+config FSL_ERRATUM_A_005337
+	bool "Work around erratum A-005337 (no hw tablewalk)"
 	depends on PPC_E500MC
+	default y
 	help
-	  Define this if you need to be able to run on rev1 silicon.
+	  This works around erratum A-005337 by not using hardware tablewalk,
+	  even if the hardware advertises it as present.
 
-          Do not define this if you are running on a simulator or rev2,
-          and want features like hardware tablewalk.
+	  Say Y if if you need to be able to run on rev1 silicon, otherwise
+	  say N for better performance.
 
 config FSL_ERRATUM_A_006184
 	bool "Work around erratum A-006184"
-- 
1.8.4.93.g57e4c17

